ChannelTypeUID on manually defined channels is not set -> NPE in ChannelItemProvider

When creating channels manually as in https://github.com/eclipse/smarthome/blob/master/docs/documentation/features/dsl.md#defining-channels, then the ChannelTypeUID of these channels is never set, and consequently, the call to ChannelItemProvider.createItemForLink() fails with an NPE because ChannelTypeRegistry.getChannelType() is getting called with null value parameter

The error thrown looks like

```
2016-06-17 10:51:05.976 [ERROR] [.c.c.registry.AbstractRegistry:154  ] - Could not inform the listener 'org.eclipse.smarthome.core.thing.internal.ChannelItemProvider$2@6cdb0d57' about the 'ADDED' event!: nulljava.lang.NullPointerException: null
	at org.eclipse.smarthome.core.thing.DefaultSystemChannelTypeProvider.getChannelType(DefaultSystemChannelTypeProvider.java:69)
	at org.eclipse.smarthome.core.thing.type.ChannelTypeRegistry.getChannelType(ChannelTypeRegistry.java:67)
	at org.eclipse.smarthome.core.thing.type.TypeResolver.resolve(TypeResolver.java:43)
	at org.eclipse.smarthome.core.thing.internal.ChannelItemProvider.getLabel(ChannelItemProvider.java:204)
	at org.eclipse.smarthome.core.thing.internal.ChannelItemProvider.createItemForLink(ChannelItemProvider.java:176)
	at org.eclipse.smarthome.core.thing.internal.ChannelItemProvider.access$1(ChannelItemProvider.java:159)
	at org.eclipse.smarthome.core.thing.internal.ChannelItemProvider$2.added(ChannelItemProvider.java:257)
	at org.eclipse.smarthome.core.thing.internal.ChannelItemProvider$2.added(ChannelItemProvider.java:1)
	at org.eclipse.smarthome.core.common.registry.AbstractRegistry.notifyListeners(AbstractRegistry.java:142)
	at org.eclipse.smarthome.core.common.registry.AbstractRegistry.notifyListeners(AbstractRegistry.java:161)
	at org.eclipse.smarthome.core.common.registry.AbstractRegistry.notifyListenersAboutAddedElement(AbstractRegistry.java:165)
```

@kaikreuzer Channels still can be created manually, or is that feature obsolete? It has a major impact on how the KNX 2.0 binding is currently working

--- a/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ChannelItemProvider.java
+++ b/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ChannelItemProvider.java
@@ -188,12 +188,13 @@ public class ChannelItemProvider implements ItemProvider {
     }
 
     private String getCategory(Channel channel) {
-        ChannelType channelType = TypeResolver.resolve(channel.getChannelTypeUID(), localeProvider.getLocale());
-        if (channelType != null) {
-            return channelType.getCategory();
-        } else {
-            return null;
+        if (channel.getChannelTypeUID() != null) {
+            ChannelType channelType = TypeResolver.resolve(channel.getChannelTypeUID(), localeProvider.getLocale());
+            if (channelType != null) {
+                return channelType.getCategory();
+            }
         }
+        return null;
     }
 
     private String getLabel(Channel channel) {
@@ -201,9 +202,11 @@ public class ChannelItemProvider implements ItemProvider {
             return channel.getLabel();
         } else {
             final Locale locale = localeProvider.getLocale();
-            final ChannelType channelType = TypeResolver.resolve(channel.getChannelTypeUID(), locale);
-            if (channelType != null) {
-                return channelType.getLabel();
+            if (channel.getChannelTypeUID() != null) {
+                final ChannelType channelType = TypeResolver.resolve(channel.getChannelTypeUID(), locale);
+                if (channelType != null) {
+                    return channelType.getLabel();
+                }
             }
         }
         return null;

